python/hidet/graph/ops/definitions/reduce/reduce_f16.py

```diff
@@ -293,6 +293,12 @@ def __init__(self, x: Tensor, dims: Optional[Sequence[int]], keepdims: bool = Fa
 def reduce_f16(
     x: Tensor, dims: Union[int, Sequence[int]], keepdims: bool, reduce_type: Union[ReduceType, str]
 ) -> Tensor:
+    from hidet.ir import Expr
+
+    if True in [isinstance(i, Expr) for i in x.shape]:
+        # TODO: either remove x.shape[-1] % 2 != 0 condition
+        # or for more longterm solution, add conditional statements to kernel launch
```

Comment:
The best solution is to add a schedule template that support x.shape[-1] % 2 == 1.

Before that, we can add the constraint in our resolve rule for reduce operator, to fall back to default implementation in case x.shape[-1] is symbolic or odd number.